### DISCUSSION TITLE
Animation controller for cut out animation

### DISCUSSION THREAD

**Discussion by sjsheelan** [opentoonz/opentoonz#6481] [Ideas]
### Is your feature request related to a problem? Please describe

There are many features that are available in opentoonz.I need animation controller to rig the puppet based on bone tools not with plastic tool.

### Describe the solution you'd like

You can expand bone tools features with some sliders with extra panel.I mean a specific panel for bone tool with slider option to change level strips with a level preview.

**Comment by RodneyBaker** [comment]
This definitely won't satisfy you but...

Opentoonz has the capability of using custom sliders even now although it's not as direct and straightforward as we would like.

Here I manipulate the position and rotation of a butterfly by adjusting the position of dial on a couple custom sliders:
![CustomSliders](https://user-images.githubusercontent.com/2576432/232455372-04696eb1-92c7-48af-8ba0-acb5e7b279c2.gif)

![CustomSliders](https://user-images.githubusercontent.com/2576432/232456036-eb7518b7-8db6-4db3-b852-5d98810875b0.gif)


It's certainly a hack but it does work.  ;)

**Comment by sjsheelan** [comment]
How can I create custom sliders?

**Comment by RodneyBaker** [comment]
What I'm using there is expressions and motion paths.
So the short answer to how is... "with great difficulty".

It might be easier to start with a question from me:

"What is it you want to apply cut out animation controllers to in your animation?"

With that direction it will be easier to tailor an answer to your question.
The real answer in my case though is:   

- Custom vector brush to create the look of the sliders and...
- Copy of the vector brush stroke to create (via copy/paste) the path the slider will travel via a motion path
- Expressions to drive the connectivity between the sliders/controllers and objects to be animated

If you are deep enough into cut out animation that you need controllers then you likely have some ability to use scripting.
This is just a different way to do that without scripting... using available features in Opentoonz to create similar functions.

Once created the controllers can be used to drive the motion of anything that can recieve an expression; columns, cameras, values of FX, etc.

**Comment by RodneyBaker** [comment]
@sjsheelan I shared these two project files on discord so I should post it here as well.

[CustomControls.zip](https://github.com/opentoonz/opentoonz/files/11261141/CustomControls.zip)
[Sliders.zip](https://github.com/opentoonz/opentoonz/files/11261142/Sliders.zip)

**Comment by sjsheelan** [comment]
Thank you ,so I have to paste those zip files ?

**Comment by RodneyBaker** [comment]
You'll need to download the files, extract the contents from the files... then open the scene file in Opentoonz.
